### PR TITLE
Add the sra-api package (OpenAPI Spec)

### PR DESCRIPTION
**Pull request by fragosti**
## Description

http://sra-api.s3-website-us-east-1.amazonaws.com/#

In https://github.com/0xProject/standard-relayer-api/pull/26 we are defining the new SRA API for 0x protocol v2. This PR will add the `sra-api` package, which will contain the Open API spec for SRA v2 along with the development tools etc...

The main idea is that we export an `api` object that is created using `@0xproject/json-schemas` etc.. and also provide an `api.json` for distribution should people need it.

Some highlights:
* Found an `OpenApiSpec` typescript type, so the OpenAPI spec gets type checks from typescript which is nice. 
* As mentioned, we are not limited to the capabilities of JSON, so can create the spec however we like: importing from `json-schemas`, using multiple files, etc...
* Set up a development environment that watches the typescript files, compiles them to JSON, and serves them up using ReDoc.
* Have a way to build and deploy a static version of the docs to s3.
* Have a way of importing markdown and including it in the spec.
* Have some basic tests for the schema validation which aren't really useful because we have types but maybe in the future we will have more complex logic to make the spec?

Currently using a fake spec, but working on implementing the actual SRA API now.

## Questions
1. Do I even need tests if the `api` object is getting typechecked?

## Plans

The idea is that we have other packages import this package for implementation:
* `sra-report` can just use an open source solution based on the Open API spec.
* `website` can just use the [ReDoc React](https://github.com/Rebilly/ReDoc#usage-as-a-react-component) component as an MVP or maybe create a custom component that knows how to render OpenAPI specs.
* `sra-client` can be a new package that distributes the bare typescript client, but also contains logic for creating and releasing the non-js / typescript packages ([codegen tool](https://github.com/OpenAPITools/openapi-generator#1---installation)).
* `connect` can depend on `sra-client` and provide some conveniences.

## Types of changes

 * New feature (non-breaking change which adds functionality)

## Checklist:

*   [x] Prefixed the title of this PR with `[WIP]` if it is a work in progress.
*   [ ] Prefixed the title of this PR with bracketed package name(s) corresponding to the changed package(s). For example: `[sol-cov] Fixed bug`.
*   [x] Added tests to cover my changes, or decided that tests would be too impractical.
*   [ ] Updated documentation, or decided that no doc change is needed.
*   [ ] Added new entries to the relevant CHANGELOG.jsons.

### CODE REVIEW

File: packages/sra-api/README.md [hunk 1]
```diff
@@ -0,0 +1,79 @@
+## @0xproject/sra-api
+
+Contains the Standard Relayer API [OpenAPI Spec](https://github.com/OAI/OpenAPI-Specification).
```
[the remark this answers is not in the document]
Comment (fragosti) [reply packages/sra-api/README.md]: Yea might make sense just to make that the README here or something? But yes should definitely use that content.

**Comment by fabioberger** [comment]
@fragosti will this package be published to NPMJS? If not, let's add the "private" prop to `package.json`.

**Comment by fragosti** [comment]
@fabioberger my expectation is that it will be published yes, so people can import a js object version of the spec via npm (we will likely do this in `website` to power the documentation via ReDoc).

File: packages/sra-api/src/examples/relayerApiAssetDataPairsResponseSchema.ts [hunk 2]
```diff
@@ -0,0 +1,21 @@
+export const relayerApiAssetDataPairsResponseSchema = {
```
[the remark this answers is not in the document]
Comment (fragosti) [reply packages/sra-api/src/examples/relayerApiAssetDataPairsResponseSchema.ts]: Why is it confusing?

File: packages/sra-api/src/api.ts [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+        title: 'Standard Relayer REST API',
+        description:
+            '0x Protocol is an open standard. Because of this, we expect many independent applications to be built that will want to use the protocol. In order to make it easier for anyone to source liquidity that conforms to the 0x order format, relayers can opt-in to implementing a set of standard relayer API endpoints. In doing so, they allow clients of the standard relayer API to access the orders on their orderbook.',
+        license: {
+            name: 'Apache 2.0',
+            url: 'https://www.apache.org/licenses/LICENSE-2.0.html',
+        },
+    },
+    servers: [
+        // TODO: Use relayer registry information here?
```
[the remark this answers is not in the document]
Comment (fragosti) [reply packages/sra-api/src/api.ts]: Yea probably just going to delete this

File: packages/sra-api/src/examples/relayerApiAssetDataPairsResponseSchema.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (BMillman19) [reply packages/sra-api/src/examples/relayerApiAssetDataPairsResponseSchema.ts]: it follows the same naming conventions as a json schema, but its an example object that conforms to some schema right?

[the remark this answers is not in the document]
Comment (fragosti) [reply packages/sra-api/src/examples/relayerApiAssetDataPairsResponseSchema.ts]: Oh oh I thought you were referring to the schema. Got it. You're right let's remove the "Schema"

**Comment by BMillman19** [comment]
Some quick notes on the ReDoc here: http://sra-api.s3-website-us-east-1.amazonaws.com/#section/Pagination

- in the pagination section of the ReDoc, the links to other sections don't work, haven't checked the other sections
- can we link out to `@0xproject/json-schemas` somewhere?
- `getOrderBook` should be `getOrderbook`

**Comment by fragosti** [comment]
ok @BMillman19 @fabioberger out of WIP

**Pull request merged**